buildscripts/testclient.py

```diff
@@ -0,0 +1,361 @@
+"""
+
+from __future__ import print_function
+
+import argparse
+from getpass import getpass
+import inspect
+import json
+import os
+from shutil import rmtree
+from subprocess import CalledProcessError, PIPE, Popen
+from tempfile import mkdtemp, mkstemp
+from textwrap import dedent
+from time import sleep, time
+from uuid import uuid4
+
+def reporoot():
+    """
+    Get the root of the git repo
+    """
+    return os.path.dirname(
+        os.path.dirname(os.path.abspath(inspect.getfile(inspect.currentframe()))))
+
+# Returns the reponame and server name
+def parse_args(args=None):
+    """
+    Parses the command line args for this command
+    """
+    parser = argparse.ArgumentParser(
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+        description=dedent("""
+            Tests the notary client against a host.
+
+            To test against a testing host without auth, just run without any arguments
+            except maybe for the server; a random repo name will be generated.
+
+
+            When running against Docker Hub, suggest usage like:
+
+                python buildscripts/testclient.py \\
+                    -s https://notary.docker.io \\
+                    -r docker.io/username/reponame \\
+                    -u username
+
+            Note that especially for Docker Hub, the repo has to have been already
+            created, else auth won't succeed.
+            """))
+    parser.add_argument(
+        '-r', '--reponame', dest="reponame", type=str,
+        help="The name of the repo - will be randomly generated if not provided")
+    parser.add_argument(
+        '-s', '--server', dest="server", type=str,
+        help="Notary Server to connect to - defaults to https://notary-server:4443")
+    parser.add_argument(
+        '-u', '--username', dest="username", type=str,
+        help="Username to use to log into the Notary Server (you will be asked for the password")
+    parsed = parser.parse_args(args)
+
+    return parsed.reponame, parsed.server, parsed.username
+
+def cleanup(*paths):
+    """
+    Best effort removal the temporary paths, whether file or directory
+    """
+    for path in paths:
+        try:
+            os.remove(path)
+        except OSError:
+            pass
+        else:
+            continue
+
+        try:
+            rmtree(path)
+        except OSError:
+            pass
+
+class Client(object):
+    """
+    Object that will run the notary client with the proper command lines
+    """
+    def __init__(self, notary_server, username_passwd=()):
+        self.notary_server = notary_server
+        self.username_passwd = username_passwd
+
+        binary = os.path.join(reporoot(), "bin", "notary")
+        self.env = os.environ.copy()
+        self.env.update({
+            "NOTARY_ROOT_PASSPHRASE": "ponies",
+            "NOTARY_TARGETS_PASSPHRASE": "ponies",
+            "NOTARY_SNAPSHOT_PASSPHRASE": "ponies",
+            "NOTARY_DELEGATION_PASSPHRASE": "ponies",
```

Comment:
Good idea!  Will make that change

On Thursday, July 14, 2016, Riyaz Faizullabhoy <notifications@github.com>
wrote:

> In buildscripts/testclient.py
> <https://github.com/docker/notary/pull/720#discussion_r70918942>:
>
> > +
> > +class Client(object):
> > +    """
> > +    Object that will run the notary client with the proper command lines
> > +    """
> > +    def __init__(self, notary_server, username_passwd=()):
> > +        self.notary_server = notary_server
> > +        self.username_passwd = username_passwd
> > +
> > +        binary = os.path.join(reporoot(), "bin", "notary")
> > +        self.env = os.environ.copy()
> > +        self.env.update({
> > +            "NOTARY_ROOT_PASSPHRASE": "ponies",
> > +            "NOTARY_TARGETS_PASSPHRASE": "ponies",
> > +            "NOTARY_SNAPSHOT_PASSPHRASE": "ponies",
> > +            "NOTARY_DELEGATION_PASSPHRASE": "ponies",
>
> nit: it'd be nice to have different passphrases just to ensure that each
> environment variable is being picked up separately. Maybe root_ponies,
> targets_ponies, etc. to keep ponies
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/docker/notary/pull/720/files/8d8229c7a853dc7c50ffa2a47ac8b1a02476a108#r70918942>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AANn4DodGO_qPG53QJ6lCW3OUn0Y4WE4ks5qVwZdgaJpZM4IYRUp>
> .
>